File: comparison/java-gps/src/main/java/TrackInfo.java
import static java.lang.System.out;

public class TrackInfo {
  public static void main(String[] args) {
    if (args.length != 1) {
      out.println("Error: filename required on command line");
      System.exit(1);
    }

    try {
      Track track = new Track(args[0]);
      out.printf("Points in track: %d\n", track.size());

      Point lowest = track.lowestPoint();
      Point highest = track.highestPoint();
      out.printf("Lowest point is %s\n", lowest);
      out.printf("Highest point is %s\n", highest);

      double distKm = track.totalDistance() / 1000.0;
      out.printf("Total distance travelled = %.3f km\n", distKm);

      double avgSpeed = track.averageSpeed();
      out.printf("Average speed = %.1f m/s\n", avgSpeed);
    }
    catch (Exception error) {
      out.println("Error: " + error.getMessage());
      System.exit(2);
    }
  }
}

File: comparison/kotlin-gps/src/main/kotlin/TrackInfo.kt
import kotlin.system.exitProcess

const val METRES_TO_KM = 0.001

fun main(args: Array<String>) {
    if (args.size != 1) {
        println("Error: filename required on command line")
        exitProcess(1)
    }

    val result = runCatching {
        with(Track(args[0])) {
            println("Points in track: $size")

            lowestPoint?.let { println("Lowest point is $it") }
            highestPoint?.let { println("Highest point is $it") }

            val distKm = totalDistance * METRES_TO_KM
            println("Total distance travelled = %.3f km".format(distKm))
            averageSpeed?.let { println("Average speed = %.1f m/s".format(it)) }
        }
    }

    if (result.isFailure) {
        println("Computation failed with an exception!")
        println(result.exceptionOrNull())
        exitProcess(2)
    }
}
